src/mongocrypt.c

```diff
@@ -1536,7 +1538,9 @@ mongocrypt_setopt_append_crypt_shared_lib_search_path (mongocrypt_t *crypt,
    // Dup the path string for us to manage
    mstr pathdup = mstr_copy_cstr (path);
    // Increase array len
+   BSON_ASSERT (crypt->opts.n_crypt_shared_lib_search_paths < INT_MAX);
    const int new_len = crypt->opts.n_crypt_shared_lib_search_paths + 1;
+   BSON_ASSERT (new_len > 0 && sizeof (mstr) <= SIZE_MAX / new_len);
    mstr *const new_array = bson_realloc (
       crypt->opts.crypt_shared_lib_search_paths, sizeof (mstr) * new_len);
```

Comment:
```suggestion
   BSON_ASSERT (new_len > 0 && sizeof (mstr) <= SIZE_MAX / (size_t) new_len);
   mstr *const new_array =
      bson_realloc (crypt->opts.crypt_shared_lib_search_paths,
                    sizeof (mstr) * (size_t) new_len);
```
Avoid implicit signed conversions.